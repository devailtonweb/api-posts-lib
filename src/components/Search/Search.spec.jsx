import { render, screen } from '@testing-library/react';
import { Search } from '.';
import userEvent from '@testing-library/user-event';

describe('<Search />', () => {
    it('should have a value of Search', () => {
        const fn = jest.fn();
        render(<Search value={'Teste valor digitado'} onChange={fn} />);

        const input = screen.getByPlaceholderText(/Type of search/i);
        expect(input).toBeInTheDocument();
        expect(input.value).toBe('Teste valor digitado');
    });

    it('should call handleChange function on each key pressed', () => {
        const fn = jest.fn();
        render(<Search value="um valor qualquer" onChange={fn} />);

        const input = screen.getByPlaceholderText(/Type of search/i);
        const value = 'o valor';

        userEvent.type(input, value);
        //screen.debug(input);
        expect(input.value).toBe('um valor qualquer');
        expect(fn).toHaveBeenCalledTimes(value.length);
    });

    it('should match snapshot', () => {
        const fn = jest.fn();
        const { container } = render(<Search value="" onChange={fn} />);

        const { firstChild } = container;
        expect(firstChild).toMatchSnapshot();
    });
});
