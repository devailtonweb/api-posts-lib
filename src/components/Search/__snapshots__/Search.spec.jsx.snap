// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should match snapshot 1`] = `
<div
  class="search-container"
>
  <input
    class="text-input"
    placeholder="Type of search"
    type="search"
    value=""
  />
</div>
`;
